lib/DynamoRequest.js

```diff
@@ -260,6 +261,15 @@ DynamoRequest.prototype.build = function () {
     delete this.data.ExpressionAttributeValues
   }
 
+  // Dynamo doesn't like it when conditions are empty
+  if (!this.data.KeyConditionExpression) {
```

Comment:
It will either be caught by the user of dynamite, or DynamoDB will throw hard errors on it.